numba/targets/arraymath.py

```diff
@@ -2535,10 +2552,10 @@ def impl(a):
             return np.nonzero(np.ravel(arr))[0]
     else:
         def impl(a):
-            if a is None:
-                data = [x for x in range(0)]
-            else:
+            if a is not None and bool(a):
                 data = [0]
+            else:
+                data = [x for x in range(0)]
```

Comment:
I think this change is a bit out of scope for this PR (not sure `argwhere` needs it?) but I think it can be accepted if appropriate unit tests are added.

Reply:
I removed the change and pushed.